include/uv/unix.h

```diff
@@ -71,6 +71,16 @@
 # include "uv/posix.h"
 #endif
 
+/* The __clang__ and __INTEL_COMPILER checks are superfluous because they
+ * define __GNUC__. They are here to convey to you, dear reader, that these
+ * macros are enabled when compiling with clang or icc.
+ */
+#if defined(__clang__) || defined(__GNUC__) || defined(__INTEL_COMPILER)
+#define UV_UNUSED(declaration) __attribute__((unused)) declaration
+#else
+#define UV_UNUSED(declaration) declaration
+#endif
```

Comment:
This shouldn't live in a public header. LMK if you want suggestions on an alternative location.

Reply:
@bnoordhuis My idea was that a header like `atomic-ops.h` should not depend on `uv-common.h` - it should be the other way around
(`internal.h` depends on `uv-common.h`)
It leaves no other choice than to create a new platform-dependent header with basic building blocks such as `UV_UNUSED` that are not to depend on `uv-common.h`?